packages/core/src/util/slotCalc.ts

```diff
@@ -31,36 +42,43 @@ export const testnetTimeSettings: TimeSettings[] = [
   { epochLength: 432_000, fromSlotDate: new Date(1_595_964_016_000), fromSlotNo: 1_598_400, slotLength: 1000 }
 ];
 
-/**
- * @returns {SlotEpochCalc} function that computes epoch # given a slot #
- */
-export const createSlotEpochCalc = (timeSettings: TimeSettings[]) => {
+const createSlotEpochCalcImpl = (timeSettings: TimeSettings[]) => {
   const timeSettingsAsc = orderBy(timeSettings, ({ fromSlotNo }) => fromSlotNo);
 
-  /**
-   * @throws TimeSettingsError
-   * @returns {Epoch} epoch of the slot
-   */
-  return (slotNo: Slot): Epoch => {
+  return (slotNo: Slot) => {
     const relevantTimeSettingsAsc = orderBy(
       timeSettingsAsc.filter(({ fromSlotNo }) => fromSlotNo <= slotNo),
       ({ fromSlotNo }) => fromSlotNo
     );
     if (relevantTimeSettingsAsc.length === 0) {
       throw new TimeSettingsError(`No TimeSettings for slot ${slotNo} found`);
     }
-    let epoch = 0;
+    let epochNo = 0;
+    let currentTimeSettings: TimeSettings;
     for (let i = 0; i < relevantTimeSettingsAsc.length; i++) {
-      const currentTimeSettings = relevantTimeSettingsAsc[i];
+      currentTimeSettings = relevantTimeSettingsAsc[i];
       const nextTimeSettings: TimeSettings | undefined = relevantTimeSettingsAsc[i + 1];
-      epoch += Math.floor(
+      epochNo += Math.floor(
         ((nextTimeSettings?.fromSlotNo || slotNo) - currentTimeSettings.fromSlotNo) / currentTimeSettings.epochLength
       );
     }
-    return epoch;
+    return { epochNo, epochTimeSettings: currentTimeSettings! };
   };
 };
 
+/**
+ * @returns {SlotEpochCalc} function that computes epoch # given a slot #
```

Comment:
```suggestion
 * @returns {SlotEpochCalc} function that computes epoch number given time settings
```

Reply:
`time settings` is an argument to this function, but this doc line is about the returned function which takes in slot number :thinking:

Reply:
Ah yes, will delete this suggestion